homeassistant/components/portainer/coordinator.py

```diff
@@ -0,0 +1,113 @@
+"""Data Updace Coordinator for Portainer."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from typing import TYPE_CHECKING
+
+from pyportainer import (
+    Portainer,
+    PortainerAuthenticationError,
+    PortainerConnectionError,
+)
+from pyportainer.models.docker import DockerContainer
+from pyportainer.models.portainer import Endpoint
+
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from . import PortainerConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=60)
+
+
+@dataclass
+class PortainerCoordinatorData:
+    """Data class for Portainer Coordinator."""
+
+    id: int
+    name: str | None
+    endpoint: Endpoint
+    containers: dict[str, DockerContainer]
+
+
+class PortainerCoordinator(DataUpdateCoordinator[dict[int, PortainerCoordinatorData]]):
+    """Data Update Coordinator for Portainer."""
+
+    portainer: Portainer
+    config_entry: PortainerConfigEntry
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: PortainerConfigEntry,
+        portainer: Portainer,
+    ) -> None:
+        """Initialize the Portainer Data Update Coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=DEFAULT_SCAN_INTERVAL,
```

Comment:
this was changed some time ago, now the config_entry has to be passed explicitly
```suggestion
            hass,
            _LOGGER,
            config_entry=config_entry,
            name=DOMAIN,
            update_interval=DEFAULT_SCAN_INTERVAL,
```